docs/plugins/plugin-development-guide/implementation.rst

```diff
@@ -0,0 +1,216 @@
+Implementation
+--------------
+
+The goal of our plugin is simple - we need to extend ``ProductVariant`` entity and provide new flag, that could be set
+on product variant form. Following customizations are done just like in the **Sylius Customization Guide**,
+take a look at :doc:`customizing models</customization/model>`, :doc:`form</customization/form>` and :doc:`template</customization/template>`.
+
+.. attention::
+
+    ``PluginSkeleton`` is focused on delivering the most friendly and testable environment. That's why in ``tests/Application`` directory,
+    there is a **tiny Sylius application** placed, with your plugin already used. Thanks to that, you can test your plugin with Behat scenarios
+    **within** Sylius application without installing it to any test app manually! There is, however, one important consequence of such an architecture.
+    **Everything** that should be done by a plugin user (configuration import, templates copying etc.) should also be done in ``tests/Application``
+    to simulate the real developer behavior - and therefore make your new features testable.
+
+Model
+*****
+
+The only field we need to add is an additional ``$availableOnDemand`` boolean. Of course we should start with the unit tests (written with
```

Comment:
Not sure it's good to use `of course` in tutorials - same as words like `simply`, `just do`, `just add`, etc. These actions are not familiar to everyone yet and might require a link or additional explanation.